### PR TITLE
Shared workflow: Publish to PyPI

### PR DESCRIPTION
**Pull request by OmarIthawi**

### CODE REVIEW

File: .github/workflows/publish-to-pypi.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: 'Publish to PyPi'
```
Comment (OmarIthawi) [inline .github/workflows/publish-to-pypi.yml:4]:
```suggestion
# This workflow will upload a Python Package using Twine when a release is created
# For more information see: https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#publishing-to-package-registries

name: 'Publish to PyPi'
```

**Comment by OmarIthawi** [comment]
@thraxil do you think this workflow would work? I'm having a trouble wrapping my head around GitHub actions `on` parameter.

I'll need to test it anyway, but trying to save myself a few bad pushes to PyPi.

**Comment by thraxil** [comment]
It should be fine. It's always hard to test these without just merging and trying it out. I think the main question on this one is whether it's worth the effort of making it into a shared workflow or if it's small enough (just a couple pretty simple steps) that it's easier to maintain just being copied into each project that needs it.

**Review by thraxil — APPROVED** [review]

**Comment by OmarIthawi** [comment]
I think copying it is fine and has worked flawlessly so far.

I needed it for the `tahoe-figures-plugins` so it looks like a shared workflow isn't the right trade-off here.

**Pull request closed without merging**